tests/scripts/all.sh

```diff
@@ -954,8 +954,8 @@ component_test_no_platform () {
     scripts/config.pl unset MBEDTLS_PSA_ITS_FILE_C
     # Note, _DEFAULT_SOURCE needs to be defined for platforms using glibc version >2.19,
     # to re-enable platform integration features otherwise disabled in C99 builds
-    make CC=gcc CFLAGS='-Werror -Wall -Wextra -std=c99 -pedantic -O0 -D_DEFAULT_SOURCE' lib programs
-    make CC=gcc CFLAGS='-Werror -Wall -Wextra -O0' test
+    make CC=gcc CFLAGS='-Werror -Wall -Wextra -std=c99 -pedantic -Os -D_DEFAULT_SOURCE' lib programs
```

Comment:
I explained why I wanted to avoid `-O0` builds in the commit message. I didn't have any particular reason to pick `-Os` vs `-O1`.

A reason not to pick `-Olarger` is that the compilation is slightly slower, OTOH a reason to pick `-Olarger` is that the tests run slightly faster. But the difference is tiny.

Reply:
Which analyses or measurements exactly are omitted when compiling with `-O0`?

Reply:
I don't know the exact list. I suspect you'd need to look at the source of individual compilers. There's some information in the [documentation of GCC warnings](https://gcc.gnu.org/onlinedocs/gcc-8.3.0/gcc/Warning-Options.html): search for `optim` to see that some warnings are less effective or disabled at `-O0`. `-Wmaybe-uninitialized` is a pretty important one that's disabled with `gcc -O0`.